wgsl/index.bs

```diff
@@ -1166,15 +1168,46 @@ allows them to naturally use values defined in the loop body.
 </div>
 * <2> The continue construct is placed at the end of the `loop`
 
-### Other Looping Constructs ### {#other-looping}
+### For statement ### {#for-statement}
 
-Note: This is *proposed* but not in the [SHORTNAME] spec yet
+<pre class='def'>
+for_stmt
+  : FOR PAREN_LEFT for_header PAREN_RIGHT BRACKET_LEFT statements BRACKET_RIGHT
+
+for_header
+  : variable_stmt? SEMICOLON logical_or_expression? SEMICOLON assignment_stmt?
```

Comment:
Why limit the statement types allowed in these contexts? Sometimes a for loop initializer is assigning a variable already declared elsewhere. Sometimes the increment is a function call. I think this is an unnecessary and harmful limitation.

Reply:
The limiting is because I didn't think of it. We can expand the initializer and continue statements, this was just the first pass at getting something in the grammar.

Does something like `(variable_stmt | assignment_stmt)?` work for the initializer? For the condition something like `(assignment_stmt | func_call_stmt)?` where `func_call_stmt` would probably need to be defined for #716 anyway.

Reply:
How about allowing any statement?

Reply:
No, the resolution from the group is that these would be kept simple.

Reply:
I asked for clarification on what "simple" is supposed to mean in "simple for loops". I wasn't;t there, but at least some folks I asked took it to mean "no `continuing` in for loops" rather than restricting the statement types. If we are to restrict it, the initializer should allow at minimum `assignment_stmt`, `variable_stmt` and the future `func_call_stment`. Other simple statement types seem less useful. Compound statements (if, loop, for itself) seem reasonable if rare though. Not clear to me if these were meant to be ruled out by the CG resolution though.

Reply:
So, if we update the initializer to `(variable_stmt | assignment_stmt | function_call_stmt)?` and the continuing to be `(assignment_stmt | func_call_stmt)?` would that work for folks?

Reply:
> No, the resolution from the group is that these would be kept simple.

I don't think the group discussed the definition of "simple".

Reply:
It didn't get recorded in the minutes, but when I said "simple for" I said that the the third clause in the for_header here should be simple, i.e. not compound.  Also if I didn't say it, I meant to similarly constrain the first clause.

I'm fine with either clause being any of of assignment_stmt, variable_stmt, func_call_stmt.